src/main/java/com/github/bibsysdev/nva/cristin/person/CristinApiClient.java

```diff
@@ -0,0 +1,190 @@
+     * Creates a NvaPerson object containing a single transformed CristinPerson. Is used for serialization to the
+     * client.
+     *
+     * @param id       The CristinPersonId of the person to query
+     * @param language Language used for some properties in the Cristin API response
+     * @return a NvaPerson filled with one transformed CristinPerson
+     * @throws BadGatewayException when there is a problem with fetch from backend
+     */
+    public NvaPerson queryOneCristinPerson(String id, String language) throws BadGatewayException {
+        CristinPerson cristinPerson = attemptToGetCristinPerson(id, language);
+        if (cristinPerson == null || !cristinPerson.hasRequiredFields()) {
+            return new EmptyNvaPerson();
+        } else {
+            NvaPerson nvaPerson = new NvaPersonBuilder(cristinPerson).build();
+            nvaPerson.setContext(PERSON_LOOKUP_CONTEXT_URL);
+            return nvaPerson;
+        }
+    }
+
+    protected static <T> T fromJson(InputStreamReader reader, Class<T> classOfT) throws IOException {
+        return OBJECT_MAPPER.readValue(reader, classOfT);
+    }
+
+    // TODO: throw BadGatewayException if this fails as well?
+    protected List<CristinPerson> queryPersons(Map<String, String> parameters) throws IOException,
+                                                                                      URISyntaxException {
+        URL url = generateQueryPersonsUrl(parameters);
+        try (InputStreamReader streamReader = getQueryResponse(url)) {
+            return asList(fromJson(streamReader, CristinPerson[].class));
+        }
+    }
+
+    protected long calculateProcessingTime(long startRequestTime, long endRequestTime) {
+        return endRequestTime - startRequestTime;
+    }
+
+    protected List<CristinPerson> queryAndEnrichPersons(Map<String, String> parameters,
+                                                        String language) throws IOException,
+                                                                                URISyntaxException {
+        List<CristinPerson> persons = queryPersons(parameters);
+        return enrichPersons(language, persons);
+    }
+
+    protected CristinPerson getPerson(String id, String language) throws IOException, URISyntaxException {
+        URL url = generateGetPersonUrl(id, language);
+        try (InputStreamReader streamReader = getResponse(url)) {
+            return fromJson(streamReader, CristinPerson.class);
+        }
+    }
+
+    protected InputStreamReader getQueryResponse(URL url) throws IOException {
+        return new InputStreamReader(url.openStream());
+    }
+
+    protected InputStreamReader getResponse(URL url) throws IOException {
+        return new InputStreamReader(url.openStream());
+    }
+
+    protected URL generateQueryPersonsUrl(Map<String, String> parameters) throws MalformedURLException,
+                                                                                 URISyntaxException {
```

Comment:
I do not see the reason to include the apache library only for the builder.
Suggestion:

```
 protected URL generateQueryPersonsUrl(Map<String, String> parameters) throws MalformedURLException,
                                                                                 URISyntaxException {
        String query=UriUtils.queryParams(parameters);
        URI uri = new URI(HTTPS, CRISTIN_API_HOST, CRISTIN_API_PERSONS_PATH, query, null);
        return uri.toURL();
    }

In UriUtils:
  public static String queryParams(Map<String, String> queryParams) {
        if(isNull(queryParams) || queryParams.isEmpty()){
            return EMPTY_STRING;
        }
        return queryParams.entrySet().stream().map(e -> String.format("%s=%s", e.getKey(), e.getValue()))
                   .collect(Collectors.joining("&"));
    }
```

Reply:
I also noticed that queryParams() method could be used in other places in the code where you construct the query parameters string.